### PR TITLE
Progress Bar

### PR DESCRIPTION
**Pull request by egregius313**
The current implementation of progress bars is not cross platform.

This set of changes replaces the current implementation of progress
bars with the package `progressbar2` which includes a full
implementation of progress-bars which can be configured if a different
output is desired.

Additionally removed a few `if type(x) == T:` type lines.

### CODE REVIEW

File: pycanvasgrader.py [hunk 1]
```diff
@@ -1181,23 +1181,27 @@ def grade_assignment(grader: PyCanvasGrader, prefs: dict):
...
         if submission.get('attachments') is not None:
-            print_on_curline('downloading submissions... ({}/{})'.format(count, total))
             if grader.download_submission(submission):
                 user_data = grader.user(user_id)
                 users.append(User(user_id, submission['id'], user_data['name'], user_data.get('email'),
                                   submission['score'], submission['grade_matches_current_submission'], submission['attempt']))
             else:
                 failed += 1
-    print_on_curline('Submissions downloaded. ({} total, {} failed to validate)\n\n'.format(total, failed))
+    os.system('cls' if os.name == 'nt' else 'clear')
```
Comment (ThePyrotechnic) [inline pycanvasgrader.py]: Switch this to use the new function

**Review by ThePyrotechnic — CHANGES_REQUESTED** [review]

File: pycanvasgrader.py (hunk 1 above)
Comment (egregius313) [reply pycanvasgrader.py]: I can definitely do that, but this was branched from the original version of master that didn't have the `clear_screen` function. So is it better to do it here without a definition of `clear_screen` or merge this and then change it?

Comment (ThePyrotechnic) [reply pycanvasgrader.py]: Rebase this branch from master, update the line, then push again

**Comment by egregius313** [comment]
Fixed the `clear_screen` issue

**Pull request closed without merging**